### PR TITLE
Clean pipeline

### PR DESCRIPTION
**Pull request by AgaMiko**

### CODE REVIEW

File: models/model.py [hunk 1]
```diff
@@ -11,24 +11,13 @@
...
 from utils.classification_mode import create_heads_dict
-
+from utils.summary_loss import SummaryLoss
 
 # initialize neptune logging
 def initialize_neptun():
     return neptune.init(api_token=NEPTUNE_API_TOKEN, project=NEPTUNE_PROJECT_NAME)
 
 
-def summary_loss(predictions, targets):
```
Comment (AgaMiko) [inline models/model.py:21]: Moved summary loss to utils

File: .gitignore [hunk 2]
```diff
@@ -141,4 +141,7 @@ dmypy.json
 .pyre/
 
 # PyCharm
-.idea/
\ No newline at end of file
+.idea/
+
+# Neptune.ai
+.neptune/
```
Comment (AgaMiko) [inline .gitignore:147]: add neptune dir to ignored files

File: models/model.py [hunk 3]
```diff
@@ -42,7 +31,7 @@ def __init__(
         num_segments=10,
         classification_mode="gloss",
         feature_extractor_name="cnn_extractor",
-        transformer_name="vanilla_transformer",
+        transformer_name="fake_transformer",
```
Comment (AgaMiko) [inline models/model.py:34]: rename `vanilla_transformer` to `fake_transformer`

File: models/model.py [hunk 4]
```diff
@@ -61,7 +50,7 @@ def __init__(
         self.num_classes_dict = create_heads_dict(classification_mode)
 
         # losses
-        # self.summary_loss = summary_loss(predictions, targets)
+        self.summary_loss = SummaryLoss(nn.CrossEntropyLoss)
```
Comment (AgaMiko) [inline models/model.py:53]: summary loss handle every loss now (**not** restricted to `nn.CrossEntropyLoss`)

File: README.md [hunk 5]
```diff
@@ -1,25 +1,53 @@
...
+## Transformer
+The transformer is a widely-used deep learning model that adopts the mechanism of self-attention, differentially weighting the significance of each part of the input data. It is used primarily in natural language processing (NLP). The Transfomer model will get representation from feature_extractor of size `num_frames,representation_size` for each video in our pipeline.
+
+## Classification heads
+Pipeline handle multihead classification. We predefine `classification_heads` for both Gloss Translation and HamNoSys recognition. Our `classification_heads` are defined here: `utils/classification_mode.py`
+
+Hamburg Sign Language Notation System (HamNoSys) is a gesture transcription alphabetic system that describes the symbols and gestures such as hand shape, hand location, and movement. Read more about HamNoSys [here - Introduction to HamNoSys](https://www.hearai.pl/post/4-hamnosys/) and [here - Introduction to HamNoSys Part 2](https://www.hearai.pl/post/5-hamnosys2/). HamNoSys always have the same number of possible classes.
+
+
+```
```
Comment (majsylw) [inline README.md:28]: this will change with new anotations of HamNoSys signs

**Review by majsylw — APPROVED** [review]

**Review by aga0 — APPROVED** [review]
LGTM

**Review by adrianlachowicz — APPROVED** [review]

**Review by mariaogryczak — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (AgaMiko) [reply .gitignore:147]: I applied it but it does not add directories to .gitignore I think. Maybe you meant requirements?

File: utils/summary_loss.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import torch
+
+
+class SummaryLoss(torch.nn.Module):
+    """ Basic loss wrapper that calcualtes multihead summary loss """
+
+    def __init__(self, loss):
```
Comment (wiktorfilipiuk) [inline utils/summary_loss.py:7]: I would consider adding typing in constructor/method definition

File: utils/summary_loss.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        """
+        Args:
+            loss (torch.nn.Module): Loss function used for multihead summary loss calculation (e.g. nn.CrossEntropyLoss)
```
Comment (wiktorfilipiuk) [inline utils/summary_loss.py:10]: The description says "loss function", param's typing says "Module". Can nn.Module be a function?

File: utils/summary_loss.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+        """
+        super().__init__()
+        self.loss = loss()
+
+    def forward(self, predictions, targets):
+        losses = []
```
Comment (wiktorfilipiuk) [inline utils/summary_loss.py:16]: Is this variable being used anywhere?
